cmd/gke-gcloud-auth-plugin/main.go

```diff
@@ -22,54 +26,81 @@ var (
 	//   email instead of numeric uniqueID.
 	defaultScopes = []string{
 		"https://www.googleapis.com/auth/cloud-platform",
-		"https://www.googleapis.com/auth/userinfo.email"}
+		"https://www.googleapis.com/auth/userinfo.email",
+	}
 )
 
-type credential struct {
-	AccessToken string                 `json:"access_token"`
-	TokenExpiry time.Time              `json:"token_expiry"`
-	X           map[string]interface{} `json:"-"` // Rest of the fields should go here.
-}
-
-// gcloudConfiguration is the struct unmarshalled
-// from gcloud config in json format
-type gcloudConfiguration struct {
-	Credential credential             `json:"credential"`
-	X          map[string]interface{} `json:"-"` // Rest of the fields should go here.
-}
+// types unmarshaled from gcloud config in json format
+type (
+	gcloudConfiguration struct {
+		Credential struct {
+			AccessToken string    `json:"access_token"`
+			TokenExpiry time.Time `json:"token_expiry"`
+		} `json:"credential"`
+		Configuration struct {
+			Properties struct {
+				Auth struct {
+					AuthorizationTokenFile string `json:"authorization_token_file"`
+				} `json:"auth"`
+			} `json:"properties"`
+		} `json:"configuration"`
+	}
+)
 
 var (
-	useAdcPtr = pflag.Bool("use_application_default_credentials", false, "returns exec credential filled with application default credentials.")
+	useApplicationDefaultCredentials = pflag.Bool("use_application_default_credentials", false, "returns exec credential filled with application default credentials.")
 )
 
 func main() {
 	pflag.Parse()
 	verflag.PrintAndExitIfRequested()
 
-	ec, err := execCredential()
+	p := plugin{
+		useApplicationDefaultCredentials: *useApplicationDefaultCredentials,
+		readGcloudConfigRaw:              readGcloudConfigRaw,
+		w:                                os.Stdout,
+	}
+
+	if err := p.run(); err != nil {
+		fmt.Fprintf(os.Stderr, "unable to retrieve access token for GKE: %s", err)
```

Comment:
Nit: start the message with a capital letter.

Reply:
Done.